deployment/ccip/changeset/internal/deploy_home_chain.go

```diff
@@ -110,6 +110,7 @@ func LatestCCIPDON(registry *capabilities_registry.CapabilitiesRegistry) (*capab
 
 // DonIDForChain returns the DON ID for the chain with the given selector
 // It looks up with the CCIPHome contract to find the OCR3 configs for the DONs, and returns the DON ID for the chain matching with the given selector from the OCR3 configs
+// TODO: this API is clunky, why a bool and an error?
 func DonIDForChain(registry *capabilities_registry.CapabilitiesRegistry, ccipHome *ccip_home.CCIPHome, chainSelector uint64) (uint32, bool, error) {
```

Comment:
let's make this error loudly if multiple dons for a chain. I'm suspicious that there are multiple dons hiding